### PR TITLE
Remove three dots menu from copy-as-to-do modal (Fixes #9980)

### LINKED ISSUE #9980: three-dots menu should not be visible in "Copy as To-Do" modal
_From Il Mago Azzurro (c073342f-4a65-4a13-9ffd-9e7fa5410d6b) in the Report a Bug guild:_

"in the Copy message as todo it's possible to open the three dots menu of a task"

screenshot_2018-02-10_20-55-31

### PR DESCRIPTION
**Pull request by hwdnoz**
Fixes #9980

### Changes
[//]: # Remove the three dots menu from the copy-as-to-do modals.

----
UUID: 
3aa5f32e-eadc-47fb-96ee-64b1022c7d43

### CODE REVIEW

**Comment by negue** [comment]
Thanks for your fix 👍

Could you also prevent the toggle on the checkbox ? 

It shouldn't work on this preview either ^^

**Comment by paglias** [comment]
It could be a good idea to create an :is-read-only prop that when set to true removes checking off the task, editing and the three dots menu. We could use it here, in the rya modal and elsewhere

**Comment by hwdnoz** [comment]
Sure, i'll add that

**Comment by Alys** [comment]
We'd need two properties: one for preventing checking off a task (for copy-as-to-do, challenge editing, challenge viewing, group plan editing, group plan viewing, but not RYA) and another for preventing editing and the three-dots menu (RYA, copy-as-todo, challenge viewing, group plan viewing, but not challenge / group plan editing).

**Comment by hwdnoz** [comment]
Sorry what does RYA modal stand for?

**Comment by hwdnoz** [comment]
need to know so i can name properties accordingly

**Comment by Alys** [comment]
Sorry! :) It's the Record Yesterday's Activity screen that you see when you start using Habitica each day. Although on the website at the moment it doesn't actually have the "Record Yesterday's Activity" title any more (it just starts with text saying "Check off any Dailies you did yesterday").

**Comment by hwdnoz** [comment]
No worries, does the last commit satisfy fixing the issue?

File: website/client/components/chat/copyAsTodoModal.vue [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <template lang="pug">
   b-modal#copyAsTodo(:title="$t('copyMessageAsToDo')", :hide-footer="true", size='md')
     .form-group
-      input.form-control(type='text', v-model='task.text')
+      input.form-control(type='text', v-model='task.text', readonly)
```
Comment (negue) [inline website/client/components/chat/copyAsTodoModal.vue:4]: I think has been a mistake, the controls inside `copyAsTodoModal` should be always editable, people want and probably will edit the text/notes before saving it

File: website/client/components/tasks/task.vue [hunk 2]
```diff
@@ -8,7 +8,7 @@
         .task-control.habit-control(:class="controlClass.up.inner", @click="(isUser && task.up) ? score('up') : null")
           .svg-icon.positive(v-html="icons.positive")
       // Dailies and todos left side control
-      .left-control.d-flex.justify-content-center(v-if="task.type === 'daily' || task.type === 'todo'", :class="controlClass.bg")
+      .left-control.d-flex.justify-content-center(v-if="(task.type === 'daily' || task.type === 'todo') && !task.preventCheck", :class="controlClass.bg")
```
Comment (negue) [inline website/client/components/tasks/task.vue:11]: I think it should be `preventCheck` not `task.preventCheck` ^^

File: website/client/components/tasks/task.vue [hunk 3]
```diff
@@ -8,7 +8,7 @@
...
         .task-control.daily-todo-control(:class="controlClass.inner", @click="isUser ? score(task.completed ? 'down' : 'up') : null")
```
Comment (negue) [inline website/client/components/tasks/task.vue:12]: the `preventCheck` should be placed here like `isUser && !preventCheck` that we still see checkbox in the preview

**Comment by hwdnoz** [comment]
would love to keep working on this, but won't be available until end of February. if someone else wants to pick it up, totally fine with me. If not, i can resume work on it in 2 weeks.

**Comment by lemoness** [comment]
@howardnoz Hi there! How's it going with this fix? Were you able to resume work?

**Comment by Alys** [comment]
@howardnoz I'm closing this and opening up the issue for other players to take it up, but if you'd like to keep working on it, we can reopen this - just let us know! :)

**Comment by hwdnoz** [comment]
O h I see. Yeah sorry about that. Thanks for managing this ticket though.

**Pull request closed without merging**